pandas/core/ops/__init__.py

```diff
@@ -499,8 +499,19 @@ def column_op(a, b):
         # in which case we specifically want to operate row-by-row
         assert right.index.equals(left.columns)
 
-        def column_op(a, b):
-            return {i: func(a.iloc[:, i], b.iloc[i]) for i in range(len(a.columns))}
+        if right.dtype == "timedelta64[ns]":
+            # ensure we treat NaT values as the correct dtype
+            # Note: we do not do this unconditionally as it may be lossy or
+            #  expensive for EA dtypes.
+            right = np.asarray(right)
+
+            def column_op(a, b):
+                return {i: func(a.iloc[:, i], b[i]) for i in range(len(a.columns))}
```

Comment:
Just to double check - is the second argument to `func` supposed to be accessed via `.iloc` here as well?

Side note - an alternate approach for by column iteration is to call `df.columns`; not sure if there is a perf difference but have seen both in code base

Reply:
> is the second argument to func supposed to be accessed via .iloc here as well?

No, at this point `b` is an ndarray.

> Side note - an alternate approach for by column iteration is to call df.columns; not sure if there is a perf difference but have seen both in code base

That runs in to difficulties if there are duplicate columns.

Reply:
> not sure if there is a perf difference

Might be worth looking at using `iat` instead of `iloc` for perf